lightning/src/ln/channelmanager.rs

```diff
@@ -4494,68 +4494,56 @@ where
 		})
 	}
 
-	fn construct_pending_htlc_status<'a>(
-		&self, msg: &msgs::UpdateAddHTLC, counterparty_node_id: &PublicKey, shared_secret: [u8; 32],
-		decoded_hop: onion_utils::Hop, allow_underpay: bool,
-		next_packet_pubkey_opt: Option<Result<PublicKey, secp256k1::Error>>,
-	) -> PendingHTLCStatus {
```

Comment:
Looks like another benefit of this refactor is that we can more easily remove the `PendingHTLCStatus` struct once its legacy usage in `channel.rs` is removed